src/daemon/plugin.c

```diff
@@ -2801,8 +2832,22 @@ int plugin_thread_create (pthread_t *thread, const pthread_attr_t *attr,
 	plugin_thread->start_routine = start_routine;
 	plugin_thread->arg           = arg;
 
-	return pthread_create (thread, attr,
+	ret = pthread_create (thread, attr,
 			plugin_thread_start, plugin_thread);
+
+	if (ret == 0 && name != NULL) {
+#if defined(HAVE_PTHREAD_SETNAME_NP) || defined(HAVE_PTHREAD_SET_NAME_NP)
+		char thread_name[16];
+		sstrncpy (thread_name, name, sizeof(thread_name));
```

Comment:
Why this copy in the first place? As @rubenk mentions, this leads to truncation and it's not clear to me why it would be necessary. It can't really be memory ownership as `thread_name` goes out of scope immediately after the `setname()` call …